ipatests/test_integration/test_dns_locations.py

```diff
@@ -83,53 +125,83 @@ def install(cls, mh):
         # give time to named to retrieve new records
         time.sleep(20)
 
-    def _test_against_server(self, server_ip, domain, expected_servers):
-        for rname, port in IPA_DEFAULT_MASTER_SRV_REC:
-            name_abs = rname.derelativize(domain)
-            expected = _gen_expected_srv_rrset(
-                name_abs, port, expected_servers)
-            query = resolve_records_from_server(
-                name_abs, 'SRV', server_ip)
-            assert expected == query, (
-                "Expected and received DNS data do not match on server "
-                "with IP: '{}' for name '{}' (expected:\n{}\ngot:\n{})".format(
-                    server_ip, name_abs, expected, query))
+    @classmethod
+    def delete_update_records(cls, rec_list=IPA_DEFAULT_MASTER_SRV_REC):
+        filepath = '/tmp/ipa.nsupdate'
+
+        cls.master.run_command([
+            'ipa', 'dns-update-system-records', '--dry-run', '--out', filepath
+        ])
+
+        if rec_list == IPA_CA_A_REC:
+            for rname in rec_list:
+                cls.master.run_command([
+                    'ipa', 'dnsrecord-del', str(cls.domain), str(rname),
+                    '--del-all'])
+
+        elif (rec_list == IPA_DEFAULT_MASTER_SRV_REC or
+              rec_list == IPA_DEFAULT_ADTRUST_SRV_REC):
+            for rname, _port in rec_list:
+                cls.master.run_command([
+                    'ipa', 'dnsrecord-del', str(cls.domain), str(rname),
+                    '--del-all'])
+
+        time.sleep(15)
+        # allow unauthenticates nsupdate (no need to testing authentication)
+        cls.master.run_command([
+            'ipa', 'dnszone-mod', str(cls.domain),
+            '--update-policy=grant * wildcard *;'
+        ], raiseonerr=False)
+
+        cls.master.run_command(['nsupdate', '-g', filepath])
+        time.sleep(15)
+
+    def _test_against_server(self, server_ip, domain, expected_servers,
```

Comment:
Please split this into 2 methods:
* _test_A_rec_....
* _test_SRV_rec_...

rather than having one big method splitted by ifs to two separated parts

Reply:
sure thing...